sql-db/sql-app/src/test/java/io/quarkus/ts/sqldb/sqlapp/XAPostgresIT.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.ts.sqldb.sqlapp;
+
+import static io.restassured.RestAssured.given;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.PostgresqlService;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.Container;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class XAPostgresIT extends AbstractSqlDatabaseIT {
+
+    static final int POSTGRESQL_PORT = 5432;
+
+    @Container(image = "${postgresql.latest.image}", port = POSTGRESQL_PORT, expectedLog = "listening on IPv4 address")
+    static PostgresqlService database = new PostgresqlService()
+            .withUser("user")
+            .withPassword("user")
+            .withDatabase("mydb");
+
+    @QuarkusApplication
+    static RestService app = new RestService().withProperties("postgresql.properties")
+            .withProperty("quarkus.datasource.username", database.getUser())
+            .withProperty("quarkus.datasource.password", database.getPassword())
+            .withProperty("quarkus.datasource.jdbc.transactions", "xa")
+            .withProperty("quarkus.datasource.jdbc.url", database::getJdbcUrl);
+
+    @Test
+    public void getSource() {
```

Comment:
There is IMO no need for a new test/endpoint, the inherited ones would do. Since we're only testing if XA transactions in native work with PostgreSQL.